lib/ansible/modules/cloud/azure/azure_rm_appserviceplan_facts.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yunge Zhu, <yungez@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_appserviceplan_facts
+
+version_added: "2.7"
+
+short_description: Get azure app service plan facts.
+
+description:
+    - Get facts for a specific app service plan or all app service plan in a resource group, or all app service plan in current subscription.
+
+options:
+    name:
+        description:
+            - Only show results for a specific app service plan.
+    resource_group:
+        description:
+            - Limit results by resource group.
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Yunge Zhu (@yungezz)"
+'''
+
+EXAMPLES = '''
+    - name: Get facts for app service plan by name
+      azure_rm_appserviceplan_facts:
+        resource_group: testrg
+        name: winwebapp1
+
+    - name: Get azure_rm_appserviceplan_facts for app service plan in resource group
+      azure_rm_webapp_facts:
+        resource_group: testrg
+
+    - name: Get facts for app service plan with tags
+      azure_rm_appserviceplan_facts:
+        tags:
+          - testtag
+          - foo:bar
+'''
+
+RETURN = '''
+azure_appserviceplan:
```

Comment:
should we remove azure_ from here? I think not needed. and for consistency with other modules: appserviceplan -> appserviceplans